### ISSUE TITLE
Multiple `inputFile.addJavaScript` calls crash build tool

### ISSUE THREAD

**Issue by klaussner** [meteor/meteor#10233]
According to [this comment](https://github.com/meteor/meteor/issues/9176#issuecomment-335913296), compiler plugins can output multiple JavaScript modules by calling `inputFile.addJavaScript` more than once:

> A compiler plugin can call `inputFile.addJavaScript` more than once for the same `inputFile`, with different output `path`, different code, different laziness, etc. This produces multiple output modules for the same source file. The output modules can be imported individually, using their specific paths, but the more common pattern is to import the original source file (that is, the `.coffee` module rather than the `.coffee.js` module).

For example, if a compiler plugin processes the file `/imports/compiler.dummy` like in the following code snippet (from [this reproduction](https://github.com/klaussner/meteor-issues/tree/10233)), two modules should be created (`/imports/compiler.dummy` and `/imports/compiler.dummy.secondModule`).

```js
var path = inputFile.getPathInPackage();

inputFile.addJavaScript({
  path: path,
  data: '"dummy 1"'
});

inputFile.addJavaScript({
  path: path + ".secondModule",
  data: '"dummy 2"'
});
```

However, Meteor crashes with this error:

```
TypeError: Cannot set property './compiler.dummy.secondModule' of undefined
    at ImportScanner._checkSourceAndTargetPaths (/tools/isobuild/import-scanner.js:442:7)
    at files.forEach.file (/tools/isobuild/import-scanner.js:271:12)
    at Array.forEach (<anonymous>)
    at ImportScanner.addInputFiles (/tools/isobuild/import-scanner.js:270:11)
    at sourceBatches.forEach.batch (/tools/isobuild/compiler-plugin.js:1284:15)
    at Array.forEach (<anonymous>)
    at Function.computeJsOutputFilesMap (/tools/isobuild/compiler-plugin.js:1255:19)
    at ClientTarget._emitResources (/tools/isobuild/bundler.js:1114:8)
    at buildmessage.enterJob (/tools/isobuild/bundler.js:841:12)
    at /tools/utils/buildmessage.js:359:18
    ...
```

The error is thrown because the import scanner attempts to add a property to the `sourceFile.deps` object but `sourceFile.deps` is `undefined`:

https://github.com/meteor/meteor/blob/c42170568af7ce798a4b12ffdc7712f3ebb5a59f/tools/isobuild/import-scanner.js#L419-L423

Tested with Meteor 1.4.4.6, 1.7.0.5, and 1.8 (:crystal_ball:).
Reproduction: https://github.com/klaussner/meteor-issues/tree/10233.

---

**Follow-up issue** (if `sourceFile.deps` wouldn't be `undefined`): If `inputFile.addJavaScript` is called only once, with a modified output path (like `path + ".secondModule"` in the example above), two modules are created: the one with the modified path contains the code (`data`) and the one with the original path re-exports everything from the first module. That way the module can be imported in the application using its original path. However, if an additional module with the original path is also created by the compiler plugin, Meteor still generates the re-export code. Is that the intended behavior?

I would like to generate multiple modules from one source file without the exports of these modules becoming exports of the module imported in the application.

**Comment by benjamn** [comment]
This should be fixed in `1.8-rc.17` (by the commit linked above). Please feel free to reopen if the problem persists!

**Comment by klaussner** [comment]
Thank you for fixing this so quickly!
